addons/binding/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting scan");
+        scheduler.execute(this::scan);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting background discovery");
+        backgroundDiscoveryFuture = scheduler.scheduleWithFixedDelay(this::scan, 0, 300, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping background discovery");
+        cancelBackgroundDiscoveryFuture();
+        super.stopBackgroundDiscovery();
+    }
+
+    private void cancelBackgroundDiscoveryFuture() {
+        if (backgroundDiscoveryFuture != null && !backgroundDiscoveryFuture.isDone()) {
+            backgroundDiscoveryFuture.cancel(true);
+            backgroundDiscoveryFuture = null;
+        }
+    }
+
+    private void scan() {
+        // Get all workgroup members
+        SmbFile[] workgroupMembers;
+        try {
+            String workgroupUrl = "smb://" + AVISUAL_WORKGROUP_NAME +"/";
+            workgroupMembers = new SmbFile(workgroupUrl).listFiles();
+        } catch (Exception e) {
+            // Can't get workgroup member list
+            return;
+        }
+
+        // Check found workgroup members for the Node devices
+        for (SmbFile s: workgroupMembers) {
+            String serverName = s.getServer();
+
+            // Check workgroup member for the Node device name match
+            Matcher m = AVISUAL_NAME_PATTERN.matcher(serverName);
+            if (!m.find()) {
+                // Workgroup member server name doesn't match the Node device name pattern
+                continue;
+            }
+
+            // Extract the Node serial number from device name
+            String nodeSerialNumber = m.group(1);
+
+            // The Node Thing UID is serial number converted to lower case
+            ThingUID thingUID = new ThingUID(AirVisualNodeBindingConstants.THING_TYPE_AVNODE,
+                    nodeSerialNumber.toLowerCase());
+
+            if (discoveryServiceCallback.getExistingDiscoveryResult(thingUID) != null ||
+                    discoveryServiceCallback.getExistingThing(thingUID) != null) {
+                // The Node with this Thing UID is already discovered or configured as a Thing, skip it
+                continue;
+            }
+
+            try {
+                // Get the Node address by name
+                NbtAddress nodeNbtAddress = NbtAddress.getByName(serverName);
+                if (nodeNbtAddress == null) {
+                    // The Node address not found by some reason, skip it
+                    continue;
+                }
+
+                // Create discovery result
+                String nodeAddress = nodeNbtAddress.getInetAddress().getHostAddress();
+                DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
+                        .withProperty(AirVisualNodeConfig.ADDRESS, nodeAddress)
```

Comment:
Since this is also unique for all devices, you should make it a `representationProperty`, see https://www.eclipse.org/smarthome/documentation/development/bindings/discovery-services.html